### PR TITLE
Add guide for synchronizing with Substrate

### PR DESCRIPTION
**Pull request by cmichi**
[Preview here](https://github.com/paritytech/substrate-contracts-node/tree/cmichi-how-to-sync#how-to-synchronize-with-substrate).

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -82,3 +84,36 @@ See our FAQ for more details:
...
+			repository with Substrate. The commit hash of the last sync is
+			mentioned in this readme.
+- [ ] Apply each commit that happened in this `node-template` folder
+      since the last sync.
+- [ ] Check commits for [`pallet-contracts`](https://github.com/paritytech/substrate/tree/master/frame/contracts)
+      since the last time someone synchronized this repository with Substrate.
+      You can limit yourself to check commits that modified the
+			`pallet-contracts` configuration in Substrate's folder `bin/node/`.
+			This folder includes the `pallet-contracts`, so whenever a breaking
+			change is introduced in the pallet it will be reflected in `bin/node/`.
```
Comment (athei) [inline README.md:100]: To what end would you check the commits?

File: README.md [hunk 2]
```diff
@@ -82,3 +84,36 @@ See our FAQ for more details:
...
+- [ ] Execute `rm Cargo.lock` for this repository.
```
Comment (athei) [inline README.md:101]: I think the canonical way would to just run:
```
cargo update -p pallet-contracts
```

It can be any crate in the substrate repo.

Comment (cmichi) [reply README.md:101]: But this then doesn't e.g. update the `Cargo.lock` to the incremented version number of `node/Cargo.toml` and `runtime/Cargo.toml`. Removing the `Cargo.lock` altogether and building it new is AFAIK the only way to account for all changes.

File: README.md (hunk 1 above)
Comment (athei) [reply README.md:100]: 👆

Comment (cmichi) [reply README.md:100]: Yeah, so it's tricky. I mean it depends on each commit. Usually you can say "just stuff that affects `bin/node/`. But there could also be things that are only relevant to this "developer" node.

Comment (cmichi) [reply README.md:100]: I left it open to interpretation, do you have a better idea how to formulate it?

**Pull request merged**